test/common/ops/changeClass.js

```diff
@@ -27,6 +28,19 @@ describe('shared.ops.changeClass', () => {
     }
   });
 
+  it('req.query.class is an invalid class', (done) => {
+    user.flags.classSelected = false;
+    user.preferences.disableClasses = false;
+
+    try {
+      changeClass(user, {query: {class: 'cellist'}});
+    } catch (err) {
+      expect(err).to.be.an.instanceof(BadRequest);
+      expect(err.message).to.equal(i18n.t('invalidClass'));
+      done();
+    }
```

Comment:
Suggestion: add an `assert(false)` or something after the `try`/`catch` block so that it will never be reached if the expected error is thrown. This will prevent the test from erroneously succeeding if the functionality is broken again in the future (causing no error is thrown and therefore no assertions either way).